### PR TITLE
Add sample for ObjectTracker skill

### PR DESCRIPTION
**Pull request by dhung-msft**

### CODE REVIEW

File: samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
+            m_descriptor = new ObjectTrackerDescriptor();
+            m_availableExecutionDevices = await m_descriptor.GetSupportedExecutionDevicesAsync();
+            if (m_availableExecutionDevices.Count == 0)
+            {
+                NotifyUser("No execution devices available, this skill cannot run on this device", NotifyType.ErrorMessage);
+                return; // Abort
+            }
+            m_skill = await m_descriptor.CreateSkillAsync() as ObjectTrackerSkill;
+            m_bindings = new List<ObjectTrackerBinding>();
+            m_trackerHistory = new List<List<TrackerResult>>();
```
Comment (dhung-msft) [inline samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs]: Consider refactoring into separate `InitializeSkillAsync` method and update README accordingly

File: samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+                                        boundingRect = binding.BoundingRect,
+                                        succeeded = true
+                                    }
+                                }
+                            );
+                        }
+
+                        m_drawnRects.Clear();
+
+                        m_bboxLock.Release();
```
Comment (dhung-msft) [inline samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs]: Consider refactoring into separate `RunSkillAsync` method and update README accordingly

File: samples/ObjectTracker/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+# Object Tracker Windows Vision Skill samples
+
+These samples will show how to use the Object Tracker Vision Skill NuGet package to create apps that can detect and classify objects in a video feed
```
Comment (LPBourret) [inline samples/ObjectTracker/README.md]: >detect and classify objects in a video feed [](start = 105, length = 43)

correction needed

File: samples/ObjectTracker/README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                        new TrackerResult()
+                        {
+                            boundingRect = binding.BoundingRect,
+                            succeeded = true
+                        }
+            }
+        );
+    }
+    m_drawnRects.Clear();
+    /* snip */
```
Comment (LPBourret) [inline samples/ObjectTracker/README.md]: this block of code needs a come comments or header that dissociate it from the eval logic in RunSkillAsync (i.e. // Initialize any new trackers selected by the user

File: samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,923 @@
...
+    /// Helper struct for storing tracker results
+    /// </summary>
+    internal struct TrackerResult
+    {
+        public Rect boundingRect;
+        public bool succeeded;
+    }
+
+    /// <summary>
+    /// Page for demonstrating the ObjectTrackerSkill on a webcam snapshot.
```
Comment (LPBourret) [inline samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs]: >on a webcam snapshot [](start = 54, length = 20)

using a webcam feed or a file

File: samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,923 @@
...
+        // Ink analyzer
+        private InkAnalyzer m_inkAnalyzer = new InkAnalyzer();
+        private List<Rect> m_drawnRects = new List<Rect>();
+
+        // Skill-related variables
+        private ObjectTrackerDescriptor m_descriptor = null;
+        private ObjectTrackerSkill m_skill = null;
+        private List<ObjectTrackerBinding> m_bindings = null;
+        private IReadOnlyList<ISkillExecutionDevice> m_availableExecutionDevices = null;
+        private List<List<TrackerResult>> m_trackerHistory = null;
```
Comment (LPBourret) [inline samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs]: >List [](start = 21, length = 4)

Use a Queue and keep a maximum amount of history (i.e. in eval, dequeue before enqueuing if your size attains this amount)

Comment (LPBourret) [reply samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs]: you only render the "pathLength" anyway, use the same constant across UI (polyline) and tracker history storage

---
In reply to: [333699935](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/59#discussion_r333699935) [](ancestors = 333699935)

**Comment by LPBourret** [comment]
may have some issue with the NBC image, we'd need a screenshot from a Creative Commons video :\

**Pull request merged**